### PR TITLE
[Storage] jumbo blobs stress tests.

### PR DESCRIPTION
**Pull request by kasobol-msft**
- Add jumbo blobs tests
- Add file share tests
- Address https://github.com/Azure/azure-sdk-for-java/issues/22579 - turns out Integer.MAX threw `java.lang.OutOfMemoryError: Requested array size exceeds VM limit`

### CODE REVIEW

File: sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java [hunk 1]
```diff
@@ -71,6 +71,8 @@
      */
     public static final int BUFFER_COPY_LENGTH = 8 * KB;
 
+    public static final int MAX_INPUT_STREAM_CONVERTER_BUFFER_LENGTH = 64 * MB;
```
Comment (gapra-msft) [inline sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java:74]: How did we come to this value? Should we add some documentation explaining why we need to do this?

**Review by gapra-msft — APPROVED** [review]

File: sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java (hunk 1 above)
Comment (kasobol-msft) [reply sdk/storage/azure-storage-common/src/main/java/com/azure/storage/common/implementation/Constants.java:74]: 1. We know that allocating Int.MAX leads to `java.lang.OutOfMemoryError: Requested array size exceeds VM limit` (there are plenty of articles about it).
2. Allocating big arrays isn't great - they put more pressure on GC if heap is fragmented and it's lesser chance of allocating one.
3. The value is guestimated based on few experiments. I.e. for jumbo blobs going to low means too much extra garbage to just wrap arrays into buffers.

**Review by alzimmermsft — APPROVED** [review]

**Pull request merged**